assisted-events-scrape/events_scrape/events_scrape.py

```diff
@@ -59,14 +66,42 @@ def run_service(self):
         if not clusters:
             log.warning("No clusters were found.")
             return None
-        with ThreadPoolExecutor(max_workers=self._max_workers) as executor:
+        with ThreadPoolExecutor(max_workers=self._max_workers) as self._executor:
```

Comment:
it's kind of problematic saving this as a class member, because after you get out of this context-manager you left with an invalid (already cleaned-up) object
so we need some way to simplify the code

Reply:
The issue here is that we need to access the executor independently from a shutdown function, in order to gracefully shutdown. Extracting this code would make it cleaner but we'd still need to access the class member in the extracted class. Even if we change strategy (injecting the actual executor at some point down the line), we still need to save reference to an object that might or might not be there. WDYT?

Reply:
Well, what I'm reading in the documentation is that you don't need to run ``shutdown()`` if you're using the context-manager (``with``): https://docs.python.org/3/library/concurrent.futures.html#concurrent.futures.Executor.shutdown

So I'd say maybe we can just do a simple assignment instead of the ``with`` usage and do proper cleanup when needed
It's also customary to set objects to ``None`` once you clean them up, so that can be an additional improvement

Reply:
Oh, I've missed that - I can remove the shutdown call here, but I was referring to signal handling shutdown, which might happen asynchronously from threadpool lifecycle https://github.com/openshift-assisted/assisted-events-scrape/pull/56/files#diff-9bf5d2c8ddd304179959d273e721b6e244a544883b8139c41abb607828378018R55